### ISSUE TITLE
Battle loss compensation is payed to the player for damaged units even if they weren't damaged in that battle

### ISSUE THREAD

**Issue by repligator** [MegaMek/mekhq#4602]
### Environment
Java 17
Linux
Master (latest PR https://github.com/MegaMek/mekhq/commit/5ab5959011ea22901321b5a4a0792da4d43d65ad)

### Severity

Low (Minor/Nuisance): Minor glitches or cosmetic issues that don’t affect gameplay and occur rarely.

### Description
The player receives battle loss compensation for damaged units at the end of a battle. However, if the player deploys a unit that is already damaged, they will still be compensated for it, as though the damage was new. There is no difference between manual and automatic resolution.

### Files
[Manual-Omega Galaxy31521015.cpnx.gz](https://github.com/user-attachments/files/16570587/Manual-Omega.Galaxy31521015.cpnx.gz)
[automatic-Omega Galaxy31521015.cpnx.gz](https://github.com/user-attachments/files/16570589/automatic-Omega.Galaxy31521015.cpnx.gz)
[mekhq.log](https://github.com/user-attachments/files/16570591/mekhq.log)

**Comment by IllianiCBT** [comment]
This will be because we're just checking whether the unit is damaged and not whether its new state differs from the state currently stored in the campaign. I'm not entirely sure how complicated this would be to resolve.

**Comment by repligator** [comment]
Yeah, it's not exactly what I'd call a high priority thing, but I'd figured I'd open an issue regardless.

**Comment by IllianiBird** [comment]
This was resolved back in August, closing
